### DISCUSSION TITLE
run custom scripts only if there is a release

### DISCUSSION THREAD

**Discussion by its-dibo** [semantic-release/semantic-release#2420] [Release workflow]
I want to run custom scripts like `npm run build` before publishing to npm or Github,  only if there is a new release

i.e: if semantic-release doesn't find a release, skip this step

**Comment by travi** [comment]
the simplest way to handle building before releasing is to define that step as a `prepack` script in your `package.json`

**Reply by travi** [reply]
> do you mean that 'prepare' will run only if schematic-release about performing a new release, otherwise it will not be run?

the `prepack` npm script runs before the `pack` phase, which happens before the `publish` step. all of these happen as part of publishing with npm and are not semantic-release specific. semantic-release only automates the process of publishing with npm, so these types of details are simply a matter of configuring npm properly

**Comment by its-dibo** [comment]
do you mean that 'prepare' will run only if schematic-release about performing a new release, otherwise it will not be run?
